core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -2016,6 +2017,37 @@ public void testSetPath()
                 .hasMessage("line 1:10: mismatched input '<EOF>'. Expecting: <identifier>");
     }
 
+    @Test
+    public void testSetTimeZone()
+    {
+        assertThat(statement("SET TIME ZONE LOCAL"))
+                .ignoringLocation()
```

Comment:
Could you please test without `ignoringLocation`?

Reply:
When not using `ignoringLocation()` the test suddenly becomes much more verbose.

```
assertThat(statement("SET TIME ZONE 'America/Los_Angeles'"))
                .ignoringLocation()
                .isEqualTo(
                        new SetTimeZone(new StringLiteral("America/Los_Angeles")));
```

becomes

```
        assertThat(statement("SET TIME ZONE 'America/Los_Angeles'"))
                .isEqualTo(
                        new SetTimeZone(
                                new NodeLocation(1, 1),
                                new StringLiteral(
                                        new NodeLocation(1, 15),
                                        "America/Los_Angeles")));
```

Do you see any added value in verbosely testing whether `antlr` works as expected?

Reply:
@martint what do you think?

Reply:
I agree with @findinpath  that with location, these tests can get very verbose. Take a look at what i recently added
https://github.com/trinodb/trino/blob/12b60f084a80c1e364e5a56db5b75ff8e917f754/core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java#L2613-L2674

OTOH, the value of tests is regression testing, and the primary cost of having a test is maintenance cost associated with keeping the code around.
I do hope the maintenance cost is not going to be high (tests change only when AST changes, but they would need to change anyway).

Also, why i would hesitate in the `SHOW STATS FOR (\n" +
                "   WITH t AS (SELECT 1 )\n" +
                "   SELECT * FROM t)` case where expect value is 56 lines, i would not hesitate here, where expected is just 5 lines. Not a big deal, is it?

Reply:
I was wrong.
In the tested scenario I'm definitely far away from saying that the test would get overly verbose.

Thank you for the suggestion. I've modified the test accordingly.